### PR TITLE
feat: add new layout component `Box`

### PR DESCRIPTION
**Pull request by IsaevAlexandr**
- Added `Box` component; Now its only can add width/height and overflow properties to the inherited component. Maybe in future there is more useful properties there;
- `Flex` component now expended from `Box`;
- `Card` component now expended from `Box` with whole set of properties;
- added `centerContent` property to `Flex` component;
- `layout components` moved in separate section in storybook;
- added info about new Box component in layout readme file;

UXRFC-267

### CODE REVIEW

File: src/components/layout/Box/Box.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    &_overflow_auto {
+        overflow: auto;
+    }
+
+    &_overflow_x {
+        overflow-x: hidden;
+        overflow-y: auto;
+    }
```
Comment (amje) [inline src/components/layout/Box/Box.scss:17]: Is it a typo? `overflow_x` mod for `overflow-y: auto`

Comment (IsaevAlexandr) [reply src/components/layout/Box/Box.scss:17]: no, `overflow_x` means that only for `x` axis content is hidden

**Review by amje — APPROVED** [review]

File: src/components/layout/LayoutProvider/__stories__/Layout.mdx [hunk 2]
```diff
@@ -157,6 +155,26 @@ export const App = () => {
 };
 ```
 
+## Box
+
+The `Box` component is a developer friend and basic block to build other components. Aware about spacing, its own sizes and most commonly used CSS properties.
```
Comment (ogonkov) [inline src/components/layout/LayoutProvider/__stories__/Layout.mdx:160]: Is this a11y friendly?

Comment (IsaevAlexandr) [reply src/components/layout/LayoutProvider/__stories__/Layout.mdx:160]: it's support all html attributes if you question about this:
https://github.com/gravity-ui/uikit/pull/1121/files#diff-048664dcf8c222b5be90577dba7bc1dfe668567e583acbe28dfdf9ec20cd0406R12-R21

**Pull request merged**